library/std/src/sys/pal/unix/net.rs

```diff
@@ -563,6 +563,24 @@ impl Socket {
         setsockopt(self, libc::SOL_SOCKET, option, mark as libc::c_int)
     }
 
+    #[cfg(any(target_os = "linux", target_os = "haiku", target_os = "vxworks"))]
+    pub fn todevice(&self) -> io::Result<&CStr> {
+        let buf: [libc::c_char; libc::IFNAMSIZ] =
+            getsockopt(self, libc::SOL_SOCKET, libc::SO_BINDTODEVICE)?;
+        let s: &[u8] = unsafe { core::slice::from_raw_parts(buf.as_ptr() as *const u8, buf.len()) };
+        let ifrname = CStr::from_bytes_with_nul(s).unwrap();
+        Ok(ifrname)
+    }
+
+    #[cfg(any(target_os = "linux", target_os = "haiku", target_os = "vxworks"))]
+    pub fn set_todevice(&self, ifrname: &CStr) -> io::Result<()> {
+        let mut buf = [0; libc::IFNAMSIZ];
+        for (src, dst) in ifrname.to_bytes().iter().zip(&mut buf[..libc::IFNAMSIZ - 1]) {
```

Comment:
this should probably check if the input name is too long rather than silently truncating it because you used `zip()`

Reply:
That value is already truncated to size specified by optlen when setsockopt is called with optlen.

Reply:
Plus length of interface is 16 on most unix systems, so no value is truncated.

Reply:
what I mean is, assuming `IFNAMSIZ == 16`, if you have an interface named `mylonginterface` (15 bytes), and you call this function with `"mylonginterfacename"` then it will silently truncate that and use `mylonginterface` instead of erroring like it should, because `setsockopt` never has a chance to see the bad name and report an error.